airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -133,12 +136,12 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 class ChildStreamMixin:
     parent_stream_class: Optional[IntercomStream] = None
 
-    def stream_slices(self, sync_mode, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+    def stream_slices(self, sync_mode: SyncMode, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
         parent_stream = self.parent_stream_class(authenticator=self.authenticator, start_date=self.start_date)
-        for slice in parent_stream.stream_slices(sync_mode=sync_mode):
+        for slice in parent_stream.stream_slices(sync_mode=sync_mode, stream_state=stream_state):
             for item in self.parent_stream_class(
                 authenticator=self.authenticator, start_date=self.start_date, stream_slice=slice
-            ).read_records(sync_mode=sync_mode):
+            ).read_records(sync_mode=sync_mode, stream_state=stream_state):
                 yield {"id": item["id"]}
```

Comment:
I completely changed my approach. I had originally introduced this change in `ChildStreamMixin` so that it filters out conversations (from parent stream) if they were updated before the current stream state. You're right in that it might introduce regression issues.

Instead of updating `ChildStreamMixin`, I made `ConversationParts` extend `HttpSubStream`, which is an implementation of `HttpStream` that handles computing of slices. That way, I don't have to modify `ChildStreamMixin` and we handle filtering/parsing response correctly. Please take a look at it in [83afd35](https://github.com/airbytehq/airbyte/pull/12374/commits/83afd357acbeb88d9333ea79ca1a38310bdf3d53).